joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -46,15 +78,33 @@ trait AstForTypesCreator(implicit withSchemaValidation: ValidationMode) { this:
       alias = None
     )
 
+    val metaTypeDeclTemp = typeDeclNode(
+      node = stmt,
+      name = s"${className}.<class>",
+      fullName = s"${classFullName}.<class>",
+      filename = relativeFileName,
+      code = code,
+      inherits = inheritsFromMeta,
+      alias = None
+    )
+
     scope.surroundingAstLabel.foreach(typeDeclTemp.astParentType(_))
     scope.surroundingScopeFullName.foreach(typeDeclTemp.astParentFullName(_))
     scope.pushNewScope(typeDeclTemp)
 
-    val bodyStmts      = astsForClassLikeBody(stmt, stmt.stmts, stmt.hasConstructor)
+    val bodyStmts      = astsForClassLikeBody(stmt, dynamicStmts, stmt.hasConstructor)
     val modifiers      = stmt.modifiers.map(modifierNode(stmt, _)).map(Ast(_))
     val annotationAsts = stmt.attributeGroups.flatMap(astForAttributeGroup)
 
     scope.popScope()
+
+    scope.surroundingAstLabel.foreach(metaTypeDeclTemp.astParentType(_))
+    scope.surroundingScopeFullName.foreach(metaTypeDeclTemp.astParentFullName(_))
```

Comment:
Is the assumption here that `surroundingAstLabel` and `surroundingScopeFullName` always return non empty?
I guess it is because otherwise `astParentType` and `astParentFullName` are not set. 
Please state this in the code be using `.get` instead of `.foreach`.